docs/configuration/http_conn_man/route_config/route.rst

```diff
@@ -153,14 +153,20 @@ Global rate limit :ref:`architecture overview <arch_overview_rate_limit>`.
 .. code-block:: json
 
   {
-    "global": "..."
+    "global": "...",
+    "rate_limit_key": "..."
   }
 
 global
   *(optional, boolean)* Specifies whether the global rate limit service should be called for a
   request that matches this route. This information is used by the :ref:`rate limit filter
   <config_http_filters_rate_limit>` if it is installed. Defaults to false if not specified.
 
+rate_limit_key
+  *(optional, string)* Specifies a descriptor value to be used for request header rate limiting.
```

Comment:
I would change this slightly to not specifically talk about request header, as we are likely to use this for other things in the future (like IP). I would just saw that the key is used by different action types and then refer to the filter docs.